src/cryptography/hazmat/primitives/kdf/kbkdf.py

```diff
@@ -51,6 +53,18 @@ def __init__(
         if not isinstance(location, CounterLocation):
             raise TypeError("location must be of type CounterLocation")
 
+        if blocation is None and location == CounterLocation.MiddleFixed:
+            raise ValueError("Please specify a blocation")
+
+        if blocation is not None and not isinstance(blocation, int):
+            raise TypeError("blocation must be an integer")
+
+        if blocation is not None and location != CounterLocation.MiddleFixed:
+            raise ValueError(
+                "blocation is ignored when location is not"
+                " CounterLocation.MiddleFixed"
+            )
+
```

Comment:
Is it necessary to check that `blocation` is within `0..len(fixed or label+context)` ?

In its present state, it's allowed to set `blocation=-1` or `=1000`. This works fine with `fixed[blocation:]` and `fixed[:blocation]` further below, but not sure how pedantic we should be.

Also not sure if you prefer to require `location=BeforeFixed` instead of allowing `location=MiddleFixed && blocation=0`, and similarly `location=AfterFixed` instead of `location=MiddleFixed && blocation=len(fixed or label+context)`.

My personal preference is to do none of these length checks and leave it up to the user, but obviously it's your call.